### PR TITLE
Add new way to create a Payment Profile. Add /client endpoints.

### PR DESCRIPTION
**Pull request by alexjohnson505**
Version 1.4.0.

Adds support for:

- Creating a new payment profile using a Braintree nonce token ```addPaymentMethodNonce```
- API /client/paymentToken - braintree's client token
- API /client/startup - startup configuration details

### CODE REVIEW

File: tests/integration/client.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+describe('Client Tests', function(){
+  describe('GET - Payment Token', function(){
+
+    it('should get the payment token', function(done){
+      tlClient.service.paymentToken(function(resp){
+        console.log(resp);
+        done();
```
Comment (officert) [inline tests/integration/client.js]: should these tests have some kind of ```assert``` or ```expect```?

File: tests/integration/payment-profile.js [hunk 2]
```diff
@@ -13,6 +13,21 @@ describe('Payment Profile', function(){
...
 	});
 
+	describe('Create with nonce', function(){
+		it('should create a payment profile with a nonce token', function(done){
+			
+			var options = {
+				paymentMethodNonce : '123',
+			}
+
+			tlPayment.service
```
Comment (officert) [inline tests/integration/payment-profile.js]: same thing for this test

Comment (alexjohnson505) [inline tests/integration/payment-profile.js]: Good point. I was having trouble running the tests. ```make tests``` didn't appear to be working.

File: src/client/service.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	.module('tl')
+	.service('tl.client.service', ['tl.service', 'tl.client.resource', function(Service, Client){
+
+		var ClientService = function(){};
+
+		/**
+		 * Generate new merchant payment token (for use with Braintree API)
+		 */
+		ClientService.prototype.paymentToken = function(success, error) {
+			return Client.paymentToken({}, success, error);
```
Comment (officert) [inline src/client/service.js]: should we return the ```.$promise``` here? I know we aren't very consistent on whether to return it or not, sometimes we do, sometimes we don't. We should try and standardize this.

Comment (AndrewBarba) [inline src/client/service.js]: I vote we return it all the time and get rid of the success and error callbacks as last two arguments for all old functions. I'd also like to update the old to take an options arguments as I find I often need to pass more things and they currently don't support it

Comment (officert) [inline src/client/service.js]: :+1:

**Comment by alexjohnson505** [comment]
Updated client endpoints to return a ```$promise```

**Comment by alexjohnson505** [comment]
@AndrewBarba Permission to merge?

File: src/index.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**
- * Tablelist.js
```
Comment (mattetre) [inline src/index.js]: this is going to get outdated real quick..

File: src/payment-profile/service.js [hunk 5]
```diff
@@ -6,7 +6,31 @@ angular
 
       var PaymentService = Service.extend(Payment);
 
+      PaymentService.prototype.addPaymentMethodNonce = function(options) {
```
Comment (mattetre) [inline src/payment-profile/service.js]: Can you add jsdocs to these two functions so the clients know how to use them

File: src/payment-profile/service.js [hunk 6]
```diff
@@ -6,7 +6,31 @@ angular
...
+          throw "An existing profile is required. Missing options.id";
+        }
+        var data = {
+          default : true
+        };
+        
+        return this.update(profileId, data).$promise;
+      };
+
       PaymentService.prototype.addPaymentProfile = function(options) {
```
Comment (mattetre) [inline src/payment-profile/service.js]: You can add a `@deprecated` to a jsdoc here to make it clear that people shouldn't use this when they are reading the docs

**Comment by mattetre** [comment]
minor comments, but looks good

File: src/payment-profile/service.js (hunk 6 above)
Comment (alexjohnson505) [inline src/payment-profile/service.js]: Good idea. Let me add *jsdoc* comments to all the payment-profile functions while I'm at it

Comment (mattetre) [inline src/payment-profile/service.js]: :+1:

**Comment by alexjohnson505** [comment]
Added **JSDocs Comments**. Marked deprecated functions as ```@deprecated```.

**Pull request merged**